[Storage] Disable live test parallelism and increase timeout

mikeharder opened the pull request:
- Add template parameters to allow individual pipelines to control parallel assembly execution and timeout.
- Storage live tests have been unreliable, and running multiple test assemblies in parallel may be contributing.
- Storage live tests have also been timing out at 3 hours, increasing to 5 hours allows tests to finish and give more useful results.
- Once tests are stable, parallel should be re-enabled (#9144) and timeout should be decreased (#9143).

mikeharder committed:
Add parallel and timeout parameters to live tests

- Allows individual pipelines to control parallel assembly execution and timeout

mikeharder commented:
/azp run net - appconfiguration - tests

mikeharder commented:
/azp run net - storage

danieljurek commented on eng/pipelines/templates/jobs/archetype-sdk-tests.yml:
```diff
@@ -2,15 +2,17 @@ parameters:
+  BuildInParallel: true
+  # Increase timeout so Event Hubs tests do not timeout (Windows runs for ~2h 35m, on average due to having two target platforms)
+  # https://github.com/Azure/azure-sdk-for-net/issues/5982
+  TimeoutInMinutes: 175
```
A trip through the last few days' scheduled tests shows that EventHubs `Test` jobs usually last around 46 minutes.. I think this comment no longer applies. 

It might make sense to set this default back down to 60. Storage tests, on the other hand, seem to consistently reach the 175 minute timeout and it might be worth upping just `sdk/storage/test.yml`'s timeout parameter.

mikeharder replied:
I'm planning to increase storage to 5 hours in a separate PR.  Once storage tests get stable, we might be able to decrease the timeout.

Let me look at the other builds to see what's reasonable for the default timeout.  And I'll remove the comment.

mikeharder committed:
Reset timeout to default of 60 minutes

mikeharder committed:
[Storage] Disable live test parallelism and increase timeout

- Storage live tests have been unreliable, and we suspect running multiple test assemblies in parallel may be contributing
- Storage live tests have also been timing out at 3 hours, increasing to 5 hours allows tests to finish and give more useful results

mikeharder commented:
/azp run net - appconfiguration - tests
